tests/components/braviatv/test_config_flow.py

```diff
@@ -0,0 +1,255 @@
+    "serial": "serial_number",
+    "macAddr": "AA:BB:CC:DD:EE:FF",
+    "name": "BRAVIA",
+    "generation": "5.2.0",
+    "area": "POL",
+    "cid": "very_unique_string",
+}
+
+BRAVIA_SOURCE_LIST = {
+    "HDMI 1": "extInput:hdmi?port=1",
+    "HDMI 2": "extInput:hdmi?port=2",
+    "HDMI 3/ARC": "extInput:hdmi?port=3",
+    "HDMI 4": "extInput:hdmi?port=4",
+    "AV/Component": "extInput:component?port=1",
+}
+
+IMPORT_CONFIG_HOSTNAME = {
+    CONF_HOST: "bravia-host",
+    CONF_PIN: "1234",
+}
+IMPORT_CONFIG_IP = {
+    CONF_HOST: "10.10.10.12",
+    CONF_PIN: "1234",
+}
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == SOURCE_USER
+
+
+async def test_import(hass):
+    """Test that the import works."""
+    with patch("bravia_tv.BraviaRC.connect", return_value=True), patch(
+        "bravia_tv.BraviaRC.is_connected", return_value=True
+    ), patch(
+        "bravia_tv.BraviaRC.get_system_info", return_value=BRAVIA_SYSTEM_INFO
+    ), patch(
+        "homeassistant.components.braviatv.async_setup_entry", return_value=True
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=IMPORT_CONFIG_HOSTNAME,
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["result"].unique_id == "very_unique_string"
+        assert result["title"] == "TV-Model"
+        assert result["data"] == {
+            CONF_HOST: "bravia-host",
+            CONF_PIN: "1234",
+            CONF_MAC: "AA:BB:CC:DD:EE:FF",
+        }
+
+
+async def test_import_cannot_connect(hass):
+    """Test that errors are shown when cannot connect to the host during import."""
+    with patch("bravia_tv.BraviaRC.connect", return_value=True), patch(
+        "bravia_tv.BraviaRC.is_connected", return_value=False
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=IMPORT_CONFIG_HOSTNAME,
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "cannot_connect"
+
+
+async def test_import_model_unsupported(hass):
+    """Test that errors are shown when the TV is not supported during import."""
+    with patch("bravia_tv.BraviaRC.connect", return_value=True), patch(
+        "bravia_tv.BraviaRC.is_connected", return_value=True
+    ), patch("bravia_tv.BraviaRC.get_system_info", side_effect=KeyError):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=IMPORT_CONFIG_IP,
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "unsupported_model"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5359319)